### PR TITLE
Wiz content pack v2.0.7: guard outgoing mirror resolve to Threat Detection issues only

### PR DESCRIPTION
**Pull request by ariel-wiz**
## Contributor
@ariel-wiz

## Summary

Fixes a bidirectional-mirroring bug where closing or resolving a mirrored incident in Cortex XSOAR attempted to set the linked Wiz Issue to **RESOLVED** regardless of the Issue's type.

In Wiz, only **Threat Detection** Issues can be manually resolved; other types (Toxic Combination, Cloud Configuration, etc.) auto-resolve once the underlying problem is remediated, and the Wiz backend rejects an explicit RESOLVED for them (`changing to resolved status is not allowed`). The outgoing-mirror paths sent RESOLVED for any type and only debug/info-logged the resulting backend rejection — so the XSOAR incident closed while the Wiz Issue silently stayed open.

The explicit `wiz-resolve-issue` command already guarded against this; this change brings the outgoing-mirror paths to parity.

## Changes

- **`Wiz.py`**
  - New `_get_issue_type` helper (returns the Issue's `type`, or `None` if not found) and `_can_resolve_issue` guard (true only for `THREAT_DETECTION`).
  - `_mirror_status_to_wiz` and `_handle_incident_closed` now skip the RESOLVED push for non-Threat-Detection Issues (letting Wiz auto-resolve) instead of firing a mutation the backend rejects and swallowing the error. `REJECTED` is unaffected (valid for all types).
  - `resolve_issue` refactored to reuse `_get_issue_type` (no behavior change).
  - `WIZ_VERSION` 1.6.0 → 1.7.0.
- **`Wiz_test.py`** — unit tests for both mirror paths × both Issue types (Threat Detection resolves; non-Threat-Detection skips), the REJECTED-still-pushes case, and an end-to-end guard through the real `_get_issue_type`.
- **Release notes** `2_0_7.md`; pack version 2.0.6 → 2.0.7.

## Testing

- `pytest Wiz_test.py` — **259 passed**.
- `ruff check` — clean. `demisto-sdk format` / `demisto-sdk validate` — all validations passed.
- Deployed to a live XSOAR 6.14 instance; code-integrity diff confirmed deployed code == source (only SDK import-stripping).
- Cross-checked against the real Wiz API (`issuesV2`) on a live tenant, both Issue types:
  - Non-Threat-Detection (Toxic Combination): guard refuses, no `updateIssue` mutation sent — XSOAR-reported type matched API ground truth.
  - Threat Detection: guard allows; resolve succeeds and API confirms `status=RESOLVED` (restored to OPEN afterward).

relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-17136

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 1]
```diff
@@ -2576,6 +2572,42 @@ def _get_issue(issue_id, is_evidence=False):
     return issue_response
 
 
+def _get_issue_type(issue_id):
+    """
```
Comment (marketplace-ai-reviewer) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:2576]: Consider adding explicit `Args` and `Returns` sections to the docstring.

File: Packs/Wiz/Integrations/Wiz/Wiz.py [hunk 2]
```diff
@@ -2576,6 +2572,42 @@ def _get_issue(issue_id, is_evidence=False):
...
+    if not isinstance(issue_object, dict):
+        return None
+    nodes = issue_object.get("data", {}).get("issues", {}).get("nodes") or []
+    if not nodes:
+        return None
+    return nodes[0].get("type")
+
+
+def _can_resolve_issue(issue_id):
+    """
```
Comment (marketplace-ai-reviewer) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:2591]: Consider adding explicit `Args` and `Returns` sections to the docstring.

File: Packs/Wiz/Integrations/Wiz/Wiz_test.py [hunk 3]
```diff
@@ -2290,6 +2292,101 @@ def test_mirror_status_to_wiz_rejected_default_reason(mock_resolve):
...
+
+    _mirror_status_to_wiz(
+        "11111111-1111-1111-1111-111111111111",
+        "resolved",
+        {"resolutionReason": "ISSUE_FIXED"},
+    )
+
+    mock_resolve.assert_not_called()
+
+
```
Comment (marketplace-ai-reviewer) [inline Packs/Wiz/Integrations/Wiz/Wiz_test.py:2348]: Consider adding a test case to verify the behavior when `_get_issue_type` returns `None`.

File: Packs/Wiz/ReleaseNotes/2_0_7.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#### Integrations
+
+##### Wiz
+
+- Fixed an issue where closing or resolving a mirrored incident in Cortex XSOAR attempted to resolve the linked Wiz Issue regardless of its type. Only Threat Detection Issues can be manually resolved; other Issue types (e.g., Toxic Combination, Cloud Configuration) auto-resolve once the underlying problem is remediated. The outgoing mirror now skips the resolve request for non-Threat-Detection Issues instead of sending a request the Wiz backend rejects.
```
Comment (marketplace-ai-reviewer) [inline Packs/Wiz/ReleaseNotes/2_0_7.md]: Please apply proper markdown styling for entities and literal values.

```suggestion
- Fixed an issue where closing or resolving a mirrored incident in Cortex XSOAR attempted to resolve the linked **Wiz** Issue regardless of its type. Only *Threat Detection* Issues can be manually resolved; other Issue types (e.g., *Toxic Combination*, *Cloud Configuration*) auto-resolve once the underlying problem is remediated. The outgoing mirror now skips the resolve request for non-*Threat Detection* Issues instead of sending a request the **Wiz** backend rejects.
```

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Wiz integration! I've reviewed the PR and found a few minor formatting and documentation items to address before we can merge. 

Specifically, please fix the `githubUser` format and add the vendor name to the keywords in the pack metadata, apply proper markdown styling to the release notes, and flesh out a couple of docstrings and a test case. Great work overall!

Additionally, please address the following file-level notes:
- `Packs/Wiz/pack_metadata.json`: - The vendor's name must appear in the `keywords` list. Please add 'Wiz' to the `keywords` array.
- The `githubUser` field must be a list of strings, but it is currently a single string. Please update it to be a list (e.g., `["ariel-wiz"]`).

 @kamalq97 please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by ariel-wiz** [comment]
Thanks for the review! Addressed in the latest commit:

- **`pack_metadata.json`** — `githubUser` is now a list (`["ariel-wiz"]`), and `"Wiz"` added to `keywords`.
- **`ReleaseNotes/2_0_7.md`** — applied the suggested markdown emphasis (`**Wiz**`, `*Threat Detection*`, `*Toxic Combination*`, `*Cloud Configuration*`).
- **`Wiz_test.py`** — added a test for the `_get_issue_type` → `None` path (issue not found ⇒ mirror skips RESOLVED), plus a direct `_get_issue_type` None test.

On the two docstring notes (`_get_issue_type` / `_can_resolve_issue` `Args`/`Returns` sections): I left these as concise one-line docstrings to stay consistent with the surrounding helpers in `Wiz.py` (e.g. `_get_issue`, `_resolve_wiz_reason`, `reopen_issue`), which don't use `Args`/`Returns` blocks. Happy to add them if you'd prefer the explicit format.

**Review by TheL0L — APPROVED** [review]
